### PR TITLE
BOAC-4084: Added notes, notes toggle, style changes to print view

### PR DESCRIPTION
**Pull request by cesarvh**
Notes, Course requirements: https://jira.ets.berkeley.edu/jira/browse/BOAC-4084 
Notes toggle: https://jira.ets.berkeley.edu/jira/browse/BOAC-4141

### CODE REVIEW

File: src/views/degree/PrintableDegreeTemplate.vue [hunk 1]
```diff
@@ -119,6 +117,14 @@
           </b-col>
         </b-row>
       </b-container>
+
+      <div v-if="template.note && includeNotes" class="ml-3">
+        <hr class="divider" />
+        <h1 id="degree-note" class="font-size-10 font-weight-bold">Degree Notes</h1>
+        <div class="font-size-8">
+          {{ template.note.body }}
```
Comment (johncrossman) [inline src/views/degree/PrintableDegreeTemplate.vue:125]: @cesarvh - Should this be `template.note`? The degree-progress-related schema is much simpler w.r.t notes.

File: src/views/degree/PrintableDegreeTemplate.vue [hunk 2]
```diff
@@ -175,13 +182,21 @@ export default {
...
   computed: {
     totalUnits() {
       // TO DO: gather completed units and subtract to get actual total units
-      return this.$_.map(this.template.unitRequirements, 'minUnits')
-        .reduce((accumulator, val) => accumulator + val)
+      const unitRequirements = this.template.unitRequirements
+
+      return unitRequirements.length? this.$_.map(this.template.unitRequirements, 'minUnits')
+        .reduce((accumulator, val) => accumulator + val) : 0
+    },
```
Comment (johncrossman) [inline src/views/degree/PrintableDegreeTemplate.vue:195]: @cesarvh - Space needed before `?`. Also, minor concern: `const unitRequirements` seems unnecessary.

**Comment by johncrossman** [comment]
Looks good @cesarvh  I made a couple suggestions.

File: src/views/degree/PrintableDegreeTemplate.vue (hunk 1 above)
Comment (cesarvh) [reply src/views/degree/PrintableDegreeTemplate.vue:125]: template.note contains other metadata (crteated/updated at/by) that we don't need to show :)

Comment (johncrossman) [reply src/views/degree/PrintableDegreeTemplate.vue:125]: Thanks @cesarvh  I missed that this is from the separate `notes` table.

**Pull request merged**